nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java

```diff
@@ -563,28 +563,66 @@ public ProcessGroupStatusSnapshotDTO clone() {
 
         other.setProcessingNanos(getProcessingNanos());
 
-        other.setConnectionStatusSnapshots(copy(getConnectionStatusSnapshots()));
-        other.setProcessorStatusSnapshots(copy(getProcessorStatusSnapshots()));
-        other.setRemoteProcessGroupStatusSnapshots(copy(getRemoteProcessGroupStatusSnapshots()));
-        other.setInputPortStatusSnapshots(copy(getInputPortStatusSnapshots()));
-        other.setOutputPortStatusSnapshots(copy(getOutputPortStatusSnapshots()));
+        if (getConnectionStatusSnapshots() != null) {
+            final List<ConnectionStatusSnapshotEntity> collectionStatusSnapshotEntities = new ArrayList<>();
+            for (final ConnectionStatusSnapshotEntity connectionStatusSnapshotEntity :  getConnectionStatusSnapshots()) {
```

Comment:
The spacing is off slightly:
```suggestion
            for (final ConnectionStatusSnapshotEntity connectionStatusSnapshotEntity : getConnectionStatusSnapshots()) {
```

Reply:
Thank you @exceptionfactory for looking at this and for the observations.
Modifications were pushed.